artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/converter/AmqpCoreConverter.java

```diff
@@ -324,7 +324,29 @@ private static ServerJMSMessage processProperties(ServerJMSMessage jms, Properti
             jms.setJMSType(properties.getSubject());
          }
          if (properties.getReplyTo() != null) {
-            jms.setJMSReplyTo(new ServerDestination(properties.getReplyTo()));
+            Byte value = (Byte)(annotations != null ? annotations.getValue().get(AMQPMessageSupport.JMS_REPLY_TO_TYPE_MSG_ANNOTATION) : AMQPMessageSupport.QUEUE_TYPE);
```

Comment:
Cast to Number, and then call .toByteValue()

Reply:
Guess What: MessageTransformationTest::testComplexQpidJMSMessageEncodeDecode was failing because of this! :)... fixing it!

Reply:
and this was the only failure on the testsuite.

Reply:
I was mistaken, I thought the JMS Client was not available at the server.

The reason we created ServerMessage was because ActiveMQMessage needed a ClientMessage and we have a ServerMessage here.


I am removing ServerDestination and using the regular ActiveMQDestination. I think the changese will become cleaner for this.


BTW: I thought this was going to e a simple bug fix.